hr_expense_invoice/models/hr_expense_sheet.py

```diff
@@ -42,7 +40,35 @@ def _compute_invoice_count(self):
                 can_read and len(sheet.expense_line_ids.mapped("invoice_id")) or 0
             )
 
-    @api.model
+    @api.depends(
+        "expense_line_ids.invoice_id.payment_state",
+        "expense_line_ids.amount_residual",
+    )
+    def _compute_payment_state(self):
+        """Determine the payment status for lines with expense invoices linked"""
+        invoice_sheets = self.filtered(lambda x: x.expense_line_ids.invoice_id)
+        invoice_sheets.payment_state = "not_paid"
+        for sheet in invoice_sheets:
```

Comment:
I think it should be filtered by status so as not to define an incorrect payment state in some cases.

Reply:
They are already filtered in the first line of the method, but all the expenses with a linked invoice should be computed in this case.